flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,61 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result((cluster, response)):
+            self.assertEqual(response, data)
+
+        def inspect_container():
+            try:
+                remote_data = run_SSH(
+                    22, 'root', data[u"host"],
+                    [
+                        b"docker",
+                        b"inspect",
+                        b"flocker--{0}".format(data[u"name"])
+                    ], None
```

Comment:
This test side-steps the public interface of the thing being tested.  We discussed some of the downsides of this kind of testing last week and tentatively agreed to stop doing things this way.

So... what is the public interface of the thing being tested?  It's ... containers.  Which is a pretty wide interface.  You could run a container that runs a web server that spits out its environment in response to a request and then you could issue a request to it.

The advantage of writing the test that way is that when we change everything around - for example, we get rid of automatic SSH access or we change the naming convention for containers - the test will still work.